src/Blocks/BlockPiston.cpp

```diff
@@ -86,28 +73,129 @@ bool cBlockPistonHandler::GetPlacementBlockTypeMeta(
 
 
 
-int cBlockPistonHandler::FirstPassthroughBlock(int a_PistonX, int a_PistonY, int a_PistonZ, NIBBLETYPE pistonmeta, cWorld * a_World)
+Vector3i cBlockPistonHandler::GetDirectionVec(int a_PistonMeta)
```

Comment:
Maybe changing the GetDirectionVec method name into MetaDataToVector name?
P.S. I've documented it.